tool/rig/main.go

```diff
@@ -119,21 +123,29 @@ func run(quiet *bool) error {
 
 	switch cmd {
 	case cupsert.FullCommand():
-		return upsert(ctx, client, config, *namespace, *cupsertChangeset, *cupsertFile)
+		if *cupsertID == "" {
+			// Concatenate all arguments and take MD5 from it as an ID
+			*cupsertID = fmt.Sprintf("%x", md5.Sum([]byte(strings.Join(os.Args[1:], ""))))
+		}
+		return upsert(ctx, client, config, *namespace, *cupsertChangeset, *cupsertFile, *cupsertID)
 	case cstatus.FullCommand():
 		return status(ctx, client, config, *namespace, *cstatusResource, *cstatusAttempts, *cstatusPeriod)
 	case cget.FullCommand():
 		return get(ctx, client, config, *namespace, *cgetChangeset, *cgetOut)
 	case cdelete.FullCommand():
-		return deleteResource(ctx, client, config, *namespace, *cdeleteChangeset, *cdeleteResourceNamespace, *cdeleteResource, *cdeleteCascade, *cdeleteForce)
+		if *cdeleteID == "" {
+			// Concatenate all arguments and take MD5 from it as an ID
+			*cdeleteID = fmt.Sprintf("%x", md5.Sum([]byte(strings.Join(os.Args[1:], ""))))
```

Comment:
Couple of thoughts here:

* I'd move this into a helper function and used here and above so the logic is not duplicated.
* I think that this way of calculating the ID has a couple of downsides:
  * It depends on the order of the command-line.
  * It only takes into account CLI args/flags. IIRC changeset ID is usually taken from an env var so it won't be accounted for here.

What if instead taking the whole command-line, we would auto-calculate ID of the operation based on a number of stable parameters: changeset name + operation type (upsert/delete) + resource namespace + resource name? This should uniquely identify a particular operation I think.